waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMappingFactoryImpl.java

```diff
@@ -91,6 +97,31 @@ private CloseableThriftHiveMetastoreIface newUnreachableMetastoreClient(Abstract
             new UnreachableMetastoreClientInvocationHandler(metaStore.getName()));
   }
 
+  private MetaStoreFilterHook loadMetastoreFilterHook(AbstractMetaStore metaStore) {
+    HiveConf conf = new HiveConf();
+    String metaStoreFilterHook = metaStore.getHiveMetastoreFilterHook();
+    if (metaStoreFilterHook == null || metaStoreFilterHook.isEmpty()) {
+      return new DefaultMetaStoreFilterHookImpl(conf);
+    }
+    Map<String, String> configurationProperties = waggleDanceConfiguration.getConfigurationProperties();
+    if (configurationProperties != null) {
+      for (Map.Entry<String, String> property : configurationProperties.entrySet()) {
+        conf.set(property.getKey(), property.getValue());
+      }
+    }
+    conf.set(HiveConf.ConfVars.METASTORE_FILTER_HOOK.varname, metaStoreFilterHook);
+    try {
+      Class<? extends MetaStoreFilterHook> filterHookClass = conf
+          .getClass(HiveConf.ConfVars.METASTORE_FILTER_HOOK.varname, DefaultMetaStoreFilterHookImpl.class,
+              MetaStoreFilterHook.class);
+      Constructor<? extends MetaStoreFilterHook> constructor = filterHookClass.getConstructor(HiveConf.class);
+      return constructor.newInstance(conf);
+    } catch (Exception e) {
+      String msg = "Unable to create instance of " + metaStoreFilterHook + ": ";
```

Comment:
```suggestion
      String message = "Unable to create instance of " + metaStoreFilterHook;
```
The message and the stack trace will be separated in the logs anyway. You could also consider inlining this on line 121 and not having the `message` variable.

Reply:
True, I just copied the `msg` thing straight from the Hive code and I didn't pay much attention to it